src/amo/components/AddonRecommendations/index.js

```diff
@@ -0,0 +1,96 @@
+/* @flow */
+import invariant from 'invariant';
+import * as React from 'react';
+import { Experiment, Variant } from 'react-ab';
+import cookie from 'react-cookie';
+import { compose } from 'redux';
+
+import tracking from 'core/tracking';
+import translate from 'core/i18n/translate';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+
+export const TAAR_IMPRESSION_CATEGORY = 'AMO Addon / Recommendations Shown';
+export const TAAR_COHORT_COOKIE_NAME = 'TAARCohort';
+export const TAAR_COHORT_INCLUDED: 'TAAR_COHORT_INCLUDED'
+  = 'TAAR_COHORT_INCLUDED';
+export const TAAR_COHORT_EXCLUDED: 'TAAR_COHORT_EXCLUDED'
+  = 'TAAR_COHORT_EXCLUDED';
+
+export type cohortName = typeof TAAR_COHORT_INCLUDED | typeof TAAR_COHORT_EXCLUDED;
+
+type Props = {|
+  _cookie: typeof cookie,
+  _randomizer: Function,
+  _tracking: typeof tracking,
+  addon: AddonType | null,
+  i18n: I18nType,
+|};
+
+
+export class AddonRecommendationsBase extends React.Component<Props> {
+  static defaultProps = {
+    _cookie: cookie,
+    _randomizer: Math.random,
+    _tracking: tracking,
+  }
+
+  componentWillMount() {
+    const { _cookie, _randomizer } = this.props;
+
+    // Set a cohort for the experiment.
```

Comment:
This appears to be working when I test it locally, but we're not 100% sure this will actually work when rendered on the server. What do you think @willdurand?

Reply:
If that works for you locally, why would it be a problem given that you should also run it on the server locally?

Reply:
ah no wait, this `componentWillMount()` method only works on the server right? So this won't work when rendered on the client if I am not mistaken.

Reply:
That doesn't appear to be the case. If I turn off SSR componentWillMount is still called. There is a note in the [docs](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount) that says "This is the only lifecycle hook called on server rendering.", but I interpret that as "out of all the lifecycle hooks, only this is available on the server", but not necessarily as "this is only available on the server".

Reply:
@willdurand Do we need to discuss or investigate this further, or do you agree with my assumptions above, based on what I observed?

Reply:
no, you're 100% right.